### ISSUE TITLE
slimmer should return species evidence from IBA evidence

### ISSUE THREAD

**Issue by nathandunn** [biolink/biolink-api#181]
TL;DR

Expand https://api.monarchinitiative.org/api/#!/bioentityset%2Fslimmer/get_entity_set_slimmer to include taxon and taxon_label. 

===

https://api.monarchinitiative.org/api/#!/bioentityset%2Fslimmer/get_entity_set_slimmer

For example for this:  https://www.alliancegenome.org/gene/WB:WBGene00015203

returns IBA evidence (among others).    However, the GO term: http://amigo.geneontology.org/amigo/term/GO:0010468

has a relevant species attached, which would be useful here, as well.  

![screen shot 2018-05-29 at 10 37 22 am](https://user-images.githubusercontent.com/751274/40675337-4defb192-632c-11e8-972a-dd8aff6e9a60.png)

The biolink call is here: 

https://api.monarchinitiative.org/api/bioentityset/slimmer/function?&slim=GO:0003824&slim=GO:0004872&slim=GO:0005102&slim=GO:0005215&slim=GO:0005198&slim=GO:0008092&slim=GO:0003677&slim=GO:0003723&slim=GO:0001071&slim=GO:0036094&slim=GO:0046872&slim=GO:0030246&slim=GO:0003674&slim=GO:0008283&slim=GO:0071840&slim=GO:0051179&slim=GO:0032502&slim=GO:0000003&slim=GO:0002376&slim=GO:0050877&slim=GO:0050896&slim=GO:0023052&slim=GO:0010467&slim=GO:0019538&slim=GO:0006259&slim=GO:0044281&slim=GO:0050789&slim=GO:0042592&slim=GO:0007610&slim=GO:0008150&slim=GO:0005576&slim=GO:0005737&slim=GO:0005856&slim=GO:0005739&slim=GO:0005634&slim=GO:0005694&slim=GO:0016020&slim=GO:0031982&slim=GO:0071944&slim=GO:0030054&slim=GO:0042995&slim=GO:0032991&slim=GO:0045202&slim=GO:0005575&subject=WB:WBGene00015203

The similar GO call is here:

http://golr.geneontology.org/solr/select?defType=edismax&qt=standard&indent=on&wt=json&rows=10&start=0&fl=*,score&facet=true&facet.mincount=1&facet.sort=count&json.nl=arrarr&facet.limit=25&hl=true&hl.simple.pre=%3Cem%20class=%22hilite%22%3E&hl.snippets=1000&fq=document_category:%22annotation%22&fq=regulates_closure:%22GO:0010468%22&facet.field=aspect&facet.field=taxon_subset_closure_label&facet.field=type&facet.field=evidence_subset_closure_label&facet.field=regulates_closure_label&facet.field=annotation_class_label&facet.field=qualifier&facet.field=annotation_extension_class_closure_label&facet.field=assigned_by&facet.field=panther_family_label&q=*:*&packet=1&callback_type=search&json.wrf=jQuery21400567218873571087_1527615692031&_=1527615692032 

Looks like the return has ```taxon`` data that we could just grab directly from the closure and the ```taxon_label```

![screen shot 2018-05-29 at 10 43 11 am](https://user-images.githubusercontent.com/751274/40675686-45870bee-632d-11e8-8375-cd9534b3b2e3.png)

**Comment by nathandunn** [comment]
FYI @cmungall  / @kltm   From the GO meeting this morning.   I think we just need to bring two additional fields through.

**Comment by nathandunn** [comment]
@deepakunni3  is this something you do when you have a moment.  Important, but not urgent.

**Comment by deepakunni3** [comment]
@nathandunn I don't think I follow.

The biolink API query is slimming GO terms associated to `WB:WBGene00015203`. Are you expecting a `taxon_label` and `taxon_label_closure` on the attached GO Terms?

The GO call highlighted above is actually a call to get just one GO term: `GO:0010468`. This call will return the term and its closure.

i.e. the calls are not similar.

**Comment by nathandunn** [comment]
@deepakunni3   Was just trying to find something close to see if the data was available with the realization that they are different terms. 

But you are right, I'm mostly looking for the `taxon` and `taxon_label`.  Closure would be fine as well, but I sense for this purpose we may not use it.

**Comment by deepakunni3** [comment]
Yes, in the results for slimmer the `taxon` property is pinned to the gene entity, which is what you normally want. The `taxon` property contains `id` and `label`. The GO term itself wouldn't have the taxon since what you are getting as result is gene -> GO term association.

Unless I am misunderstanding the use-case here.

**Comment by selewis** [comment]
It's a misunderstanding. Yes, we do get a taxon property for the gene that
has been annotated.

However, in many cases the basis for the annotation is from a second gene
(what is in the GAF 'with' column) and what is returned from biolink in the
"evidence__with" property. For example, the follow query:
https://api.monarchinitiative.org/api/bioentityset/slimmer/function?&slim=GO:0003824&slim=GO:0004872&slim=GO:0005102&slim=GO:0005215&slim=GO:0005198&slim=GO:0008092&slim=GO:0003677&slim=GO:0003723&slim=GO:0001071&slim=GO:0036094&slim=GO:0046872&slim=GO:0030246&slim=GO:0003674&slim=GO:0008283&slim=GO:0071840&slim=GO:0051179&slim=GO:0032502&slim=GO:0000003&slim=GO:0002376&slim=GO:0050877&slim=GO:0050896&slim=GO:0023052&slim=GO:0010467&slim=GO:0019538&slim=GO:0006259&slim=GO:0044281&slim=GO:0050789&slim=GO:0042592&slim=GO:0007610&slim=GO:0008150&slim=GO:0005576&slim=GO:0005737&slim=GO:0005856&slim=GO:0005739&slim=GO:0005634&slim=GO:0005694&slim=GO:0016020&slim=GO:0031982&slim=GO:0071944&slim=GO:0030054&slim=GO:0042995&slim=GO:0032991&slim=GO:0045202&slim=GO:0005575&subject=MGI:MGI:97490

returns

   - evidence_with:
   [
      - "MGI:MGI:1856155"
      ],
   And we have to cobble together (based on the prefix) a guess of the
   taxon for this gene. You'd likely have to do something similar since no one
   is actually providing this information. When it is a UniProt identifier it
   may be human, or sea anemone, or elk, or ...


Particularly for human it would be nice to get the taxon.

**Comment by nathandunn** [comment]
We *could* do it that way, but I think that the there is a taxon for that specific evidence, as well, which is displayed on the AmiGO page I showed further down.  But, maybe I misunderstood.

Nathan
